dev/com.ibm.ws.kernel.service/src/com/ibm/ws/common/crypto/CryptoUtils.java

```diff
@@ -404,8 +412,21 @@ public static boolean isSemeruFips() {
         return "true".equals(getPropertyLowerCase("semeru.fips", "false"));
     }
 
+    protected static boolean useEnhancedSecurityAlgorithms() {
+        if (isEnhancedSecurityChecked) {
+            return isEnhancedSecurity;
+        } else {
+            isEnhancedSecurity = isRunningBetaMode() && Boolean.valueOf(getPropertyLowerCase(PROPERTY_USE_ENHANCED_SECURITY_ALG, "false"));
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "isEnhancedSecurity: " + (isEnhancedSecurity ? "enabled" : "disabled"));
+            }
+        }
+        isEnhancedSecurityChecked = true;
+        return isEnhancedSecurity;
+    }
+
     public static boolean isFips140_3EnabledWithBetaGuard() {
-        return isRunningBetaMode() && isFips140_3Enabled();
+        return (isRunningBetaMode() && isFips140_3Enabled()) || useEnhancedSecurityAlgorithms();
```

Comment:
i don't think we need to add it here because we are already calling it inside `isFips140_3Enabled` and `useEnhancedSecurityAlgorithms` already has a beta guard in it.

also i plan to remove or comment out this method when we GA semeru fips.

Reply:
Makes sense, originally i didn't have the beta check in useEnhancedSecurityAlgorithms so I added it in the return statement.